structure/heap/heap.go

```diff
@@ -0,0 +1,79 @@
+package heap
+
+// The Interface type describes the requirements
+// for a type using the routines in this package.
+type Interface[T any] interface {
+	Less(o T) bool
+}
```

Comment:
I do not think you need this interface... The type constraint `constraints.Ordered` in the constraints package within this repo is what you need I think.

Reply:
We can do this like testing example 2 if we have this interface, but the `constraints.Ordered` can't.

Reply:
Thank you so much for your detailed reply, I benefited a lot from this commit. For the two options you mentioned, I provide two methods `New` and `NewAny`, users can choose according to their needs.